File: src/main/java/com/tjh/riskfactor/entity/json/ApiError.java
package com.tjh.riskfactor.entity.json;

import lombok.Setter;
import lombok.Getter;

import com.fasterxml.jackson.annotation.JsonFormat;

import java.util.Date;

@Getter @Setter
public class ApiError {

    @JsonFormat(shape = JsonFormat.Shape.NUMBER)
    private Date timestamp;

    private Integer status;

    private String error;

    @Setter
    private String message;

    private String uri;

    private String body = null;

    private String stacktrace = null;

}

File: backend/src/main/java/com/tjh/riskfactor/error/ApiError.kt
package com.tjh.riskfactor.error

import com.fasterxml.jackson.annotation.JsonFormat

import java.util.Date

data class ApiError(
    val status: Int
) {
    @JsonFormat(shape = JsonFormat.Shape.NUMBER)
    val timestamp: Date = Date()

    var error: String = ""
    var message: String = ""
    var uri: String = ""

    var body: String? = null
    var stacktrace: String? = null

}
